### DISCUSSION TITLE
Is semaphore made for team work ?

### DISCUSSION THREAD

**Discussion by ySp-chld** [semaphoreui/semaphore#1114] [Ideas]
Maybe it is a discussion and not an issue but here's my point:
- ssh key/token/password are shared among a project.  
- there's no way to use only the user ssh key and you're forced to run project against ssh key set in template without editing template.  
- auditing is not the best yet but it will become I'm sure  
- there's no way to my knowledge to create groups in user management.  
- inventory management is super messy if using internal editor it needs to be a proper text editor with syntax coloring and proper sized frame. and if using files there's a need to explore accessible files from a repo and map it to where it will be store inside the server. Just having to know the precise path where a repo will be put by Semaphore is not really a solution.  
- also a ssh key shouldn't be linked to an inventory.  

Here's what I would think about in the future to share a project as a team:  
- Possibility to set a key/token/password per user and only use said ssh key when running a template from ssh and github and vault eventually.  
- a proper inventory editor or a simple way to select a file from a repo or list available file in repo.

Those are only though on my mind at the moment I don't know if reasonable or shared by others, but at the moment I feel like Semaphore is not a tool to be share but a tool you spawn on your computer to run playbooks from your computer.  

Now maybe an example to illustrate the workflow I imagine : 

- an admin create a playbooks to update software on pool of server.
- the inventory is inside a git repo
- a vault with password for the application in inside the playbook
- a template run is created by the admin 
- when a user connect to the project if he click on run the vault key should be accessible to him only if admin has shared it.
- the playbook should run in the user name with his credential and ssh key.
- no edit has occurs by the user only the click on a button and eventually some survey answer.
I don't see how semaphore could be used in a professional environment without such capabilities.

Am i misunderstanding the way such tool should be used or maybe if semaphore was design with this in mind ?
maybe I do not see how a team should use such tool but if only the person who created the template can run it with his credential and he need to share the task and the credential on production machine this is an issue IMHO.

What do you think about it ?

**Comment by peteeckel** [comment]
In addition in an enterprise environment it would be very helpful (or rather, essential) to be able to assign project team memberships using LDAP groups.
